lib/action_controller/serialization.rb

```diff
@@ -57,11 +57,14 @@ def use_adapter?
     [:_render_option_json, :_render_with_renderer_json].each do |renderer_method|
       define_method renderer_method do |resource, options|
         options.freeze
-        serializer_options = options.deep_dup
-        serializer_options.fetch(:serialization_context) do
-          serializer_options[:serialization_context] = ActiveModelSerializers::SerializationContext.new(request, serializer_options)
+        serialization_options = options.deep_dup
+        if options.key?(:serializer)
+          serialization_options[:serializer] = options[:serializer]
```

Comment:
you said there was some weird bug caused by duping the `serializer` that this fixes?